src/environment.cpp

```diff
@@ -271,6 +271,238 @@ ABMWithState::ABMWithState(ActiveBlockModifier *abm_):
 }
 
 /*
+	LBMManager
+*/
+
+LBMContentMapping::~LBMContentMapping()
+{
+	for (std::vector<LoadingBlockModifierDef *>::iterator it = lbm_list.begin();
+			it != lbm_list.end(); ++it) {
+		delete *it;
+	}
+}
+
+void LBMContentMapping::addLBM(LoadingBlockModifierDef *lbm_def, IGameDef *gamedef)
+{
+	// Add the lbm_def to the LBMContentMapping.
+	// Unknown names get added to the global NameIdMapping.
+	INodeDefManager *nodedef = gamedef->ndef();
+
+	lbm_list.push_back(lbm_def);
+
+	for (std::set<std::string>::const_iterator it = lbm_def->trigger_contents.begin();
+			it != lbm_def->trigger_contents.end(); ++it) {
+		std::set<content_t> c_ids;
+		bool found = nodedef->getIds(*it, c_ids);
+		if (!found) {
+			content_t c_id = gamedef->allocateUnknownNodeId(*it);
+			if (c_id == CONTENT_IGNORE) {
+				// Seems it can't be allocated.
+				warningstream << "Could not internalize node name \"" << *it
+					<< "\" while loading LBM \"" << lbm_def->name << "\"." << std::endl;
+				continue;
+			}
+			c_ids.insert(c_id);
+		}
+
+		for (std::set<content_t>::const_iterator iit =
+				c_ids.begin(); iit != c_ids.end(); ++iit) {
+			content_t c_id = *iit;
+			container_map::iterator jt = map.find(c_id);
+			if (jt == map.end()) {
+				std::vector<LoadingBlockModifierDef *> def_list;
+				map[c_id] = def_list;
+				jt = map.find(c_id);
+			}
+			jt->second.push_back(lbm_def);
+		}
+	}
+}
+
+const std::vector<LoadingBlockModifierDef *>
+		*LBMContentMapping::lookup(content_t c) const
```

Comment:
In this case it's probably better to put the `*` on the same line as `std::vector`, otherwise the return type is confusing.
The reasoning for putting the `*` next to the value doesn't even apply here anyway (the reasoning is that `char* a, b;` is equivalent to `char* a; char b;` not `char* a; char* b;` as you'd expect, but functions can only have one return value so it doesn't apply).